src/modules/users/users.repository.ts

```diff
@@ -0,0 +1,46 @@
+import { db } from "database/data-source";
+import { User } from "shared/entities";
+import { DatabaseError, NotFoundError } from "shared/errors";
+import { UsersCreateDto } from "./users.types";
+
+export class UsersRepository {
```

Comment:
So you are not using typeorm ?

Reply:
Yes, I hope it's ok? I come from a NoSQL background and this time decided to make this a challenge and try writing pure SQL while learning it